codex/blockexchange/engine/discovery.nim

```diff
@@ -143,8 +143,6 @@ proc start*(b: DiscoveryEngine) {.async.} =
     asyncSpawn fut
 
   b.discoveryLoop = b.discoveryQueueLoop()
-  b.trackedFutures.track(b.discoveryLoop)
-  asyncSpawn b.discoveryLoop
 
 proc stop*(b: DiscoveryEngine) {.async.} =
```

Comment:
Since you are changing `b` => `self` all over the places, maybe you want to do it here as well?

Reply:
kinda wanted to keep the noise down, it was probably a mistake to have done that in the engine, but I couldn't help myself :)